util/util.go

```diff
@@ -204,6 +204,14 @@ func GetClusterByName(ctx context.Context, c client.Client, namespace, name stri
 	return cluster, nil
 }
 
+// ClusterKey returns NamespacedName for the cluster object
+func ClusterKey(cluster *clusterv1.Cluster) client.ObjectKey {
```

Comment:
Don't have to make this change here in this PR, but we could rename this to ObjectKey and have it take in an
```go
type ObjectKeyer interface {
  GetNamespace() string
  GetName() string
}
```

Then it would be applicable to more than just Clusters.

Reply:
Initially was looking into utilizing this helper, which implements described functionality. https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/interfaces.go#L33-L39 The only problem, is an additional error handling.

Reply:
Went ahead and did a change on the PR. Hope it will be helpful later.